build_engine_test_images/terraform/aws/ubuntu/main.tf

```diff
@@ -0,0 +1,221 @@
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+  access_key = var.build_engine_aws_access_key
+  secret_key = var.build_engine_aws_secret_key
+}
+
+# Create a random string suffix for instance names
+resource "random_string" "random_suffix" {
+  length           = 8
+  special          = false
+  lower            = true
+  upper            = false
+}
+
+# Create a VPC
+resource "aws_vpc" "build_engine_aws_vpc" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name = "${var.build_engine_aws_vpc_name}-${random_string.random_suffix.id}"
+  }
+}
+
+# Create internet gateway
+resource "aws_internet_gateway" "build_engine_aws_igw" {
+  vpc_id = aws_vpc.build_engine_aws_vpc.id
+
+  tags = {
+    Name = "build_engine_igw"
+  }
+}
+
+# Create controlplane security group
+resource "aws_security_group" "build_engine_aws_secgrp" {
+  name        = "build_engine_aws_secgrp"
+  description = "Allow all inbound traffic"
+  vpc_id      = aws_vpc.build_engine_aws_vpc.id
+
+  ingress {
+    description = "Allow SSH"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "build_engine_aws_sec_grp_controlplane-${random_string.random_suffix.id}"
+  }
+}
+
+# Create Public subnet
+resource "aws_subnet" "build_engine_aws_public_subnet" {
+  vpc_id     = aws_vpc.build_engine_aws_vpc.id
+  availability_zone = var.aws_availability_zone
+  cidr_block = "10.0.1.0/24"
+
+  tags = {
+    Name = "build_engine_public_subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create route table for public subnets
+resource "aws_route_table" "build_engine_aws_public_rt" {
+  depends_on = [
+    aws_internet_gateway.build_engine_aws_igw,
+  ]
+
+  vpc_id = aws_vpc.build_engine_aws_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.build_engine_aws_igw.id
+  }
+
+  tags = {
+    Name = "build_engine_aws_public_rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Assciate public subnet to public route table
+resource "aws_route_table_association" "build_engine_aws_public_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.build_engine_aws_public_subnet,
+    aws_route_table.build_engine_aws_public_rt
+  ]
+
+  subnet_id      = aws_subnet.build_engine_aws_public_subnet.id
+  route_table_id = aws_route_table.build_engine_aws_public_rt.id
+}
+
+# Create AWS key pair
+resource "aws_key_pair" "build_engine_aws_pair_key" {
+  key_name   = format("%s_%s", "build_engine_aws_key_pair", "${random_string.random_suffix.id}")
+  public_key = file(var.aws_ssh_public_key_file_path)
+}
+
+# Create controlplane instances
+resource "aws_instance" "build_engine_aws_instance" {
+ depends_on = [
+    aws_subnet.build_engine_aws_public_subnet,
+  ]
+
+  count = var.build_engine_aws_instance_count
```

Comment:
Do we need multiple instances to build images?

Reply:
Currently 1 is enough

Reply:
I mean maybe we don't need this count?

Reply:
I am more prefer kept this count variable, this part of code actually already reviewed by former infra @meldafrawi